server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -273,100 +273,96 @@ public void createSnapshotLegacy(final CreateSnapshotRequest request, final Acti
         final SnapshotId snapshotId = new SnapshotId(snapshotName, UUIDs.randomBase64UUID()); // new UUID for the snapshot
         Repository repository = repositoriesService.repository(request.repository());
         final Map<String, Object> userMeta = repository.adaptUserMetadata(request.userMetadata());
-        clusterService.submitStateUpdateTask(
-            "create_snapshot [" + snapshotName + ']',
-            new ClusterStateUpdateTask(request.masterNodeTimeout()) {
-
-                private List<String> indices;
-
-                private SnapshotsInProgress.Entry newEntry;
+        repository.executeConsistentStateUpdate(repositoryData -> new ClusterStateUpdateTask(request.masterNodeTimeout()) {
```

Comment:
Should we pass this `repositoryData` down to `beginSnapshot` to avoid having to fetch it again later and potentially better consistency?

Reply:
It'll (almost certainly) be read from cache there so the fetching is nbd. I'm not 100% sure about the consistency argument, `beginSnapshot` dispatches back to `SNAPSHOT` so are we sure nothing might change it in the meantime? I'm inclined not to change things in this area without a strong motivation at least.

Reply:
> I'm not 100% sure about the consistency argument, beginSnapshot dispatches back to SNAPSHOT so are we sure nothing might change it in the meantime?

Thanks for explaining. Now I noticed that the legacy `initializingSnapshots` behaviour which leads to thread dispatching. You are right that we don't want to just reuse the `repositoryData`. Also agree to not change things unless absolutely necessary.